### PR TITLE
[ECS][Rector][PHPMD][PHPStan] Update quality tools

### PR DESCRIPTION
**Pull request by ERuban**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | -   

BC: 
- some sniffers or rector have been deleted (by maintainers). 
- configs should be updated according documentations (ours and official)

Updates:
- Update ECS to the latest version
- Update custom sniffs and its tests to support latest ECS
- Update rector to the latest possible version
- Update rectors and its tests to support latest rector
- Update documentation
- Add sets

### CODE REVIEW

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: .github/workflows/php-ci.yml [hunk 1]
```diff
@@ -34,7 +35,78 @@ jobs:
...
                 run: |
                     composer install --prefer-dist --no-progress --no-scripts
+    rector-tests:
+        needs: composer
+        runs-on: ubuntu-latest
+        timeout-minutes: 60
+        strategy:
+            matrix:
+                php: ['7.4']
+        name: Run rectors tests
```
[the remark this answers is not in the document]
Comment (ERuban) [reply .github/workflows/php-ci.yml]: done

File: .github/workflows/php-ci.yml [hunk 2]
```diff
@@ -34,7 +35,78 @@ jobs:
...
+                run: |
+                    vendor/bin/phpunit tests/Rector
+    sniff-tests:
+        needs: composer
+        runs-on: ubuntu-latest
+        timeout-minutes: 60
+        strategy:
+            matrix:
+                php: ['7.4']
+        name: Run sniffs tests
```
[the remark this answers is not in the document]
Comment (ERuban) [reply .github/workflows/php-ci.yml]: Done

File: .github/workflows/php-ci.yml [hunk 3]
```diff
@@ -34,7 +35,78 @@ jobs:
...
+                    php-version: ${{ matrix.php }}
+                    coverage: none
+
+            -   name: Get the cached Composer dependencies
+                uses: eonx-com/actions-cache@v1
+                with:
+                    path: vendor
+                    key: ${{ matrix.php }}-composer-${{ hashFiles('composer.lock') }}
+
+            -   name: Run Sniff tests
```
[the remark this answers is not in the document]
Comment (ERuban) [reply .github/workflows/php-ci.yml]: done

File: .github/workflows/php-ci.yml [hunk 4]
```diff
@@ -34,7 +35,78 @@ jobs:
...
+                uses: eonx-com/actions-cache@v1
+                with:
+                    path: vendor
+                    key: ${{ matrix.php }}-composer-${{ hashFiles('composer.lock') }}
 
-            -   name: Run tests
+            -   name: Run other tests
                 run: |
-                    vendor/bin/phpunit tests
+                    vendor/bin/phpunit tests/Output
```
[the remark this answers is not in the document]
Comment (ERuban) [reply .github/workflows/php-ci.yml]: done

File: docs/ecs/settings.md [hunk 5]
```diff
@@ -21,25 +22,23 @@ return static function (ContainerConfigurator $containerConfigurator): void {
...
     ]);
     
-    $parameters->set(Option::EXCLUDE_PATHS, [
-        __DIR__ . '/path/to/file.php',
-        __DIR__ . '/path/with/mask/**/*.php',
-    ]);
-    
     $services = $containerConfigurator->services();
     
+    // To add rules you want or to override rules from sets.
```
[the remark this answers is not in the document]
Comment (ERuban) [reply docs/ecs/settings.md]: done

File: docs/rectors/settings.md [hunk 6]
```diff
@@ -52,9 +49,7 @@ return static function (ContainerConfigurator $containerConfigurator): void {
 
     $services = $containerConfigurator->services();
 
-    $services->set(StrictInArrayRector::class);
-    $services->set(ThrowWithPreviousExceptionRector::class);
-    $services->set(ExplicitBoolCompareRector::class);
+    // To add rules you want or to override rules from sets.
```
[the remark this answers is not in the document]
Comment (ERuban) [reply docs/rectors/settings.md]: done

File: tests/Rector/AddSeeAnnotationRector/Fixture/MissingTag.php.inc [hunk 7]
```diff
@@ -54,7 +54,7 @@ class MissingTag extends TestCase
      * Provides data.
      *
      * @return mixed[]
-     *
+     *
```
[the remark this answers is not in the document]
Comment (ERuban) [reply tests/Rector/AddSeeAnnotationRector/Fixture/MissingTag.php.inc:57]: So, as discussed in Slack - we can't do it at this moment, `PHPStan\PhpDocParser` creates new empty line in PhpDoc with this space symbol.

**Pull request merged**